Microsoft.Toolkit.Uwp.UI.Controls/MarkdownTextBlock/Helpers/SmartWeakEvent.cs

```diff
@@ -0,0 +1,174 @@
+﻿// ******************************************************************
```

Comment:
Correct me if I'm wrong, but the WeakEventListener looks like something the consumer would have to do. The smart weak events allow the consumer to register just like normal events, but not have to worry about removing the registration when they clean up. 

I could remove the SmartWeakEvent logic and just make them normal events, but then the consumer would have to use the WeakEventListener to get the same effect.